istioctl/cmd/tag.go

```diff
@@ -0,0 +1,400 @@
+	cmd.AddCommand(tagListCommand())
+	cmd.AddCommand(tagRemoveCommand())
+
+	return cmd
+}
+
+func tagSetCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:        "set",
+		Short:      "Create or modify revision tags",
+		Example:    "istioctl x tag set prod --revision 1-8-0",
+		SuggestFor: []string{"create"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a tag for modification")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("can only provide a single tag for creation")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+
+			return setTag(context.Background(), client.Kube(), args[0], revision, cmd.OutOrStdout())
+		},
+	}
+
+	cmd.PersistentFlags().BoolVar(&overwrite, "overwrite", false, "whether to overwrite an existing tag")
+	cmd.PersistentFlags().BoolVarP(&skipConfirmation, "skip-confirmation", "y", false, "whether to skip confirmation for tag deletion")
+	cmd.PersistentFlags().StringVarP(&revision, "revision", "r", "", "revision to point tag to")
+	_ = cmd.MarkPersistentFlagRequired("revision")
+
+	return cmd
+}
+
+func tagListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List existing revision tags and their corresponding revisions",
+		Example: "istioctl x tag list",
+		Aliases: []string{"show"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("tag list command does not accept arguments")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+			return listTags(context.Background(), client.Kube(), cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+func tagRemoveCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "remove",
+		Short:   "Remove an existing revision tag",
+		Example: "istioctl x tag remove prod",
+		Aliases: []string{"delete"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a tag for removal")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("must provide a single tag for removal")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+
+			return removeTag(context.Background(), client.Kube(), args[0], skipConfirmation, cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+// setTag creates or modifies a revision tag.
+func setTag(ctx context.Context, kubeClient kubernetes.Interface, tag, revision string, w io.Writer) error {
+	revWebhooks, err := getWebhooksWithRevision(ctx, kubeClient, revision)
+	if err != nil {
+		return err
+	}
+	if len(revWebhooks) == 0 {
+		return fmt.Errorf("cannot find webhook with revision %q", revision)
```

Comment:
These errors will be exposed to users because users will misspell tags or become confused about the difference between revisions and revision tags.  So work on the Errors to make them clear.  E.g.  "cannot modify tag: no Istio injection MutatingWebhookConfiguraiton for Istio revision tag %q".  Do this for any Error that will be generated by typos.  (Less care is needed for things like parsing a Webhook after someone hand-edits it with "kubectl edit".)